src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -151,11 +166,41 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
 
   try {
     checkTezosConnection()
-    mainLoop(0)
+    val accountsRefreshEpochs = Await.result(unprocessedEpochsForRefreshingAccounts(), atMost = 5.seconds)
+    mainLoop(0, SortedSet(accountsRefreshEpochs: _*))
   } finally {
     shutdown()
   }
 
+  /* Possibly updates all accounts if the current block level is past any of the epochs
+   * @param epochs the relevant epochs that calls for a refresh
+   */
+  private def processAccountRefreshes(epochs: SortedSet[Int]): Future[SortedSet[Int]] =
+    if (epochs.nonEmpty) {
+      for {
+        storedHead <- apiOperations.fetchMaxLevel
+        updated <- if (epochs.exists(_ <= storedHead)) {
+          val (past, toCome) = epochs.partition(_ <= storedHead)
+          logger.info(
+            "A block level was reached requiring to update the accounts data. Relevant block levels are {}",
```

Comment:
Can we change this to: The current block, $block, requires an update of account data as specified in the config file. A full refresh is now underway.

Reply:
I changed the sentence to mostly correspond to your suggestions.
I'm not keen on referring to a "current block", because the check will be made between lorre cycles (*), meaning that we might have already passed the relevant block by a significant margin (and even more than one actually, but only the most recent refresh will take effect).

(*) this additionally implies that the refresh effect will not be generally apparent until a full cycle is completed, which might be a long run when bootstrapping from blank.